python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -76,6 +76,8 @@ class OpRuntimeMetrics:
     num_tasks_have_outputs: int = field(default=0, metadata={"map_only": True})
     # Number of finished tasks.
     num_tasks_finished: int = field(default=0, metadata={"map_only": True})
+    # Number of failed tasks.
+    num_tasks_failed: int = field(default=0, metadata={"map_only": True})
```

Comment:
Keep using the term "tasks" here for consistency with other metrics.

Reply:
Can we add this to the `Dataset.stats()` output?